### PR TITLE
Update version of cloudcache lib we reference

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/54869

### CODE REVIEW

File: src/VisualStudio/CSharp/Test/PersistentStorage/AbstractPersistentStorageTests.cs [hunk 1]
```diff
@@ -30,9 +29,12 @@ public enum Size
         {
             Small,
             Medium,
-            Large
+            Large,
+            ExtraLarge,
         }
 
+        private const int Iterations = 100;
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/CSharp/Test/PersistentStorage/AbstractPersistentStorageTests.cs:36]: note: even running all tests 100 times each, we still run all sqlite and cloudcache tests in under 30s total.  so this seems worth it for the coverage.

File: eng/Versions.props [hunk 2]
```diff
@@ -116,7 +116,7 @@
     <MicrosoftServiceHubClientVersion>2.8.10</MicrosoftServiceHubClientVersion>
     <MicrosoftServiceHubFrameworkVersion>2.8.10</MicrosoftServiceHubFrameworkVersion>
     <MicrosoftVisualBasicVersion>10.1.0</MicrosoftVisualBasicVersion>
-    <MicrosoftVisualStudioCacheVersion>16.10.40-alpha</MicrosoftVisualStudioCacheVersion>
+    <MicrosoftVisualStudioCacheVersion>17.0.13-alpha</MicrosoftVisualStudioCacheVersion>
```
Comment (dibarbet) [inline eng/Versions.props:119]: might need to go into main-vs-deps unless we're shipping this in VS, I don't think we've merged dev17 back into main yet (though hopefully soon)

Comment (CyrusNajmabadi) [reply eng/Versions.props:119]: main-vs-deps already has this.  this is to unblock main through from teh test failures we're seeing there.  do you know when we're going to be on dev17 for main?  @jinujoseph ?

Comment (dibarbet) [reply eng/Versions.props:119]: I believe we're waiting on preview2 images? @allisonchou was working on it actively if I remember right

Comment (CyrusNajmabadi) [reply eng/Versions.props]: @dibarbet can this be merged in now?

Comment (dibarbet) [reply eng/Versions.props]: Yep I think so!

**Review by dibarbet — APPROVED** [review]

**Pull request merged**